src/com/owncloud/android/lib/common/network/FileRequestEntity.java

```diff
@@ -124,14 +127,39 @@ public void writeRequest(final OutputStream out) throws IOException {
                     }
                 }
             }
-            
+
         } catch (IOException io) {
-            Log_OC.e("FileRequestException", io.getMessage());
-            throw new RuntimeException("Ugly solution to workaround the default policy of retries when the server falls while uploading ; temporal fix; really", io);   
-            
+            // any read problem will be handled as if the file is not there
+            if (io instanceof FileNotFoundException) {
+                throw io;
+            } else {
+                FileNotFoundException fnf = new FileNotFoundException("Exception reading source file");
+                fnf.initCause(io);
+                throw fnf;
+            }
+
+        } catch (WriteException we) {
+            throw we.getWrapped();
+
         } finally {
-            channel.close();
-            raf.close();
+            try {
+                channel.close();
+                raf.close();
+            } catch (IOException io) {
```

Comment:
Maybe could be cleanest check if Files are null to close them instead of a try-catch treatment.